frontend/src/lib/stores/SidebarExpandedStore.ts

```diff
@@ -1,11 +1,32 @@
 import { writable } from 'svelte/store';
+import { browser } from '$app/environment';
 
 function createIsSidebarOpenStore() {
-    const { subscribe, update } = writable<boolean>(true);
+    let sidebar;
+
+    if (browser && localStorage.getItem('isSidebarOpen') === null) {
+        sidebar = writable<boolean>(true);
+    } else {
+        sidebar = writable<boolean>(
+            (browser && localStorage.getItem('isSidebarOpen') === 'true') || false
+        );
+    }
+
+    const sidebarExpanded = sidebar;
+
+    sidebarExpanded.subscribe((value) => {
+        if (browser) return localStorage.setItem('isSidebarOpen', String(value));
+    });
+
+    const toggle = () => {
+        console.log('toggling!');
+        sidebarExpanded.update((value) => !value);
+    };
```

Comment:
great publisher/subscriber pattern here. You might need to remove the console.log